frontend/src/stores/searchParams/index.ts

```diff
@@ -92,23 +97,69 @@ export const useSimpleSearchParamsStore = defineStore(
     const setSort = (value: string) => (sort.value = value);
 
     function $reset() {
-      const initialState = getInitialState(router);
+      reinitializeFromQuery(route.query);
+    }
+
+    function reinitializeFromQuery(routerQuery: LocationQuery) {
+      const initialState = getInitialState(routerQuery);
 
       query.value = initialState.query;
       pageNumber.value = initialState.pageNumber;
       category.value = initialState.category;
       itemsPerPage.value = initialState.itemsPerPage;
       sort.value = initialState.sort;
       court.value = initialState.court;
-      dateParams.$reset();
+      dateParams.reset(initialState);
     }
 
     /*
     ##############
     #  watchers  #
     ##############
-*/
-    watch(params, () => setRouterQuery(router, params.value));
+    */
+
+    /*
+     track what query has been set by the store, in order to prevent infinite
+     update loops
+    */
+    const routerQuerySetByStore = ref(omitDefaults(params.value));
+
+    const postHogStore = usePostHogStore();
+    const updateRouterQuery = (router: Router, params: LocationQueryRaw) => {
+      const query = omitDefaults(params);
+      const previousQuery = routerQuerySetByStore.value;
+      postHogStore.searchPerformed(
+        "simple",
+        addDefaults(query),
+        addDefaults(previousQuery),
+      );
+      routerQuerySetByStore.value = query;
+      return router.push({
+        ...route,
+        query,
+      });
+    };
+
+    watch(params, () => updateRouterQuery(router, params.value));
+
+    watch(
+      () => route.query,
+      async (newQuery) => {
+        // check whether the store already has the values, in order to prevent loops
+        const updatedQueryWasSetByStore = _.isEqual(
+          _.mapValues(
+            toRaw(routerQuerySetByStore.value),
+            (value) => value?.toString(), // numbers in query are represented as strings
+          ),
+          newQuery,
+        );
+
+        if (!updatedQueryWasSetByStore) {
+          reinitializeFromQuery(newQuery);
+          console.log(newQuery);
+        }
+      },
+    );
```

Comment:
To me, the piece of code took a little while to understand and it needed comments to explain.
I would simplify it and make small functions to explain what happens and make it easier to debug and give single responsibility to each method of the code. Here's my suggestion or something in the same line of thought:

```suggestion
function normalizeQuery(query: Record<string, any>): Record<string, string> {
  return Object.fromEntries(
    Object.entries(query).map(([k, v]) => [k, v?.toString() ?? ""])
  );
}

function isSameQuery(
  storeQuery: Record<string, any>,
  routeQuery: Record<string, any>
): boolean {
  return _.isEqual(
    normalizeQuery(toRaw(storeQuery)),
    normalizeQuery(routeQuery)
  );
}

watch(
  () => route.query,
  (newQuery) => {
    if (!isSameQuery(routerQuerySetByStore.value, newQuery)) {
      reinitializeFromQuery(newQuery);
      console.log("The query was set by the user and is equal to:", newQuery);
    }
  }
);
```

Reply:
Good call, I've refactored the code to make the intention clearer.